pkg/metrics/metrics.go

```diff
@@ -0,0 +1,180 @@
+	secrets corelisters.SecretNamespaceLister
+	config  sampoplisters.ConfigLister
+}
+
+func (sc *samplesCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- importsFailedDesc
+	ch <- invalidSecretDesc
+}
+
+func (sc *samplesCollector) Collect(ch chan<- prometheus.Metric) {
+	skips := map[string]bool{}
+	cfg, err := sc.config.Get(configv1.ConfigName)
+	if err != nil {
+		logrus.Infof("metrics sample config retrieval failed with: %s", err.Error())
+		return
+	}
+	for _, skip := range cfg.Spec.SkippedImagestreams {
+		skips[skip] = true
+	}
+
+	importFailuresExist := cfg.ConditionTrue(configv1.ImportImageErrorsExist)
+	importFailures := cfg.Condition(configv1.ImportImageErrorsExist)
+	importFailuresReason := importFailures.Reason
+	for _, stream := range Streams {
+		_, skipped := skips[stream]
+		dontCare := cfg.Status.ManagementState == operatorv1.Removed || cfg.Status.ManagementState == operatorv1.Unmanaged
+
+		if skipped || dontCare {
+			// unmanaged/removed/skipping-an-imagestream means we consider the metric as 0 since we don't care if it failed import
+			addCountGauge(ch, importsFailedDesc, stream, float64(0))
+			continue
+		}
+		// if there are import failures, and this stream is in the list maintained in the Reason field, set to 1
+		if importFailuresExist && (strings.HasPrefix(importFailuresReason, stream+" ") || strings.Contains(importFailuresReason, " "+stream+" ") || strings.HasSuffix(importFailuresReason, " "+stream)) {
+			addCountGauge(ch, importsFailedDesc, stream, float64(1))
+			continue
+		}
+		addCountGauge(ch, importsFailedDesc, stream, float64(0))
+	}
+
+	if !cfg.ConditionTrue(configv1.ConfigurationValid) {
+		configInvalidStat.Set(1)
+	} else {
+		configInvalidStat.Set(0)
+	}
+
+	if !cfg.ConditionTrue(configv1.ImportCredentialsExist) {
+		addCountGauge(ch, invalidSecretDesc, missingSecret, float64(1))
+	} else {
+		addCountGauge(ch, invalidSecretDesc, missingSecret, float64(0))
+	}
+	secret, err := sc.secrets.Get(configv1.SamplesRegistryCredentials)
+	if err != nil {
+		logrus.Infof("metrics pull secret retrieval failed with: %s", err.Error())
+		addCountGauge(ch, invalidSecretDesc, missingTBRCredential, float64(1))
+		return
+	}
+	dockerConfigJSON := dockerConfigJSON{}
+	buf, ok := secret.Data[".dockerconfigjson"]
+	if !ok {
+		logrus.Infof(".dockerconfigjson is missing from pull secret")
+		addCountGauge(ch, invalidSecretDesc, missingTBRCredential, float64(1))
+		return
+	}
+	err = json.Unmarshal(buf, &dockerConfigJSON)
+	if err != nil {
+		logrus.Infof("error unmarshalling .dockerconfigjson: %s", err.Error())
+		addCountGauge(ch, invalidSecretDesc, missingTBRCredential, float64(1))
+		return
+	}
+	_, hasTBRCreds := dockerConfigJSON.Auths["registry.redhat.io"]
+	if hasTBRCreds {
+		addCountGauge(ch, invalidSecretDesc, missingTBRCredential, float64(0))
+		return
+	}
+	logrus.Printf(".dockerconfigjson is missing an entry for registry.redhat.io")
+	addCountGauge(ch, invalidSecretDesc, missingTBRCredential, float64(1))
```

Comment:
per above, i don't think we can confidently report this as an error/alert.

Reply:
see my above response on checking registry override first